### ISSUE TITLE
Reuse evaluation environments instead of recreating them every cycle

### ISSUE THREAD

**Issue by TTomilin** [TTomilin/MEAL#4]
> **File:** `baselines/IPPO_CL.py` 

### 📋 Problem

The current evaluation logic repeatedly constructs new environments every time `evaluate_model()` is called:

```python
def evaluate_model(train_state, key):
    envs = pad_observation_space()
    for eval_idx, env in enumerate(envs):
        env = make(config.env_name, layout=env)  # 💥 slow + redundant
```

---

### 🐢 Additional inefficiencies

Aside from env recreation, this code relies on:
- **Python `for` loops** over environments (not `jax.lax.scan` or `vmap`)
- **Side-effect-heavy `make()` calls** within a JIT scope (bad for performance + reproducibility)
- **Dynamic padding and index adjustment logic** executed every evaluation

These break the vectorized design of the rest of the training pipeline.

---
